File: Problem 1 Codes/1_chatgpt_sol2.java
import java.io.*;
import java.util.*;

class Main {

    static boolean canMove(long[] arr, long x) {
        long first = arr[0];

        if (first == x) {
            return true;
        }

        long[] sorted = arr.clone();
        Arrays.sort(sorted);

        int firstPos = -1;
        int xPos = -1;

        for (int i = 0; i < sorted.length; i++) {
            if (sorted[i] == first) {
                firstPos = i;
            }

            if (sorted[i] == x) {
                xPos = i;
            }
        }

        int left = Math.min(firstPos, xPos);
        int right = Math.max(firstPos, xPos);

        for (int i = left; i < right; i++) {
            if (sorted[i + 1] - sorted[i] != 1) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine().trim());

        while (t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());

            int n = Integer.parseInt(st.nextToken());
            long x = Long.parseLong(st.nextToken());

            long[] arr = new long[n];

            st = new StringTokenizer(br.readLine());

            for (int i = 0; i < n; i++) {
                arr[i] = Long.parseLong(st.nextToken());
            }

            System.out.println(canMove(arr, x) ? "YES" : "NO");
        }
    }
}

File: Problem 1 Codes/1_chatgpt_sol2.cpp
#include <bits/stdc++.h>
using namespace std;

bool canMoveToFront(vector<long long> arr, long long x)
{
    long long front = arr[0];

    if (front == x)
        return true;

    sort(arr.begin(), arr.end());

    int frontPos = -1;
    int targetPos = -1;

    for (int i = 0; i < (int)arr.size(); i++)
    {
        if (arr[i] == front)
            frontPos = i;

        if (arr[i] == x)
            targetPos = i;
    }

    int left = min(frontPos, targetPos);
    int right = max(frontPos, targetPos);

    for (int i = left; i < right; i++)
    {
        if (arr[i + 1] - arr[i] != 1)
            return false;
    }

    return true;
}

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;

    while (T--)
    {
        int N;
        long long X;

        cin >> N >> X;

        vector<long long> A(N);

        for (auto &value : A)
            cin >> value;

        if (canMoveToFront(A, X))
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}